### PR TITLE
Feature/day care opportunities

### PR DESCRIPTION
**Pull request by dunnudge**
## Describe this PR

Move day care opportunities to a separate entity

### *What changes have we introduced*

- Created new day care opportunities entity
- Implement create, read and update for the entity
- Allow creating day care opportunities when creating the day care package or stand alone

### CODE REVIEW

File: LBH.AdultSocialCare.Api/V1/Boundary/DayCarePackageOpportunityBoundary/Request/DayCarePackageOpportunityForUpdateRequest.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+namespace LBH.AdultSocialCare.Api.V1.Boundary.DayCarePackageOpportunityBoundary.Request
+{
+    public class DayCarePackageOpportunityForUpdateRequest
```
Comment (petemade) [inline LBH.AdultSocialCare.Api/V1/Boundary/DayCarePackageOpportunityBoundary/Request/DayCarePackageOpportunityForUpdateRequest.cs:3]: I wonder if the creation and update objects can be merged to DayCarePackageOpportunityRequest?

File: LBH.AdultSocialCare.Api/V1/Domain/DayCarePackageDomains/DayCarePackageDomain.cs [hunk 2]
```diff
@@ -23,9 +23,6 @@ public class DayCarePackageDomain
         public bool TransportNeeded { get; set; }
         public bool EscortNeeded { get; set; }
         public int TermTimeConsiderationOptionId { get; set; }
-        public string HowLong { get; set; }
-        public string HowManyTimesPerMonth { get; set; } // Daily, weekly, monthly
-        public string OpportunitiesNeedToAddress { get; set; }
         public DateTimeOffset DateCreated { get; set; } = DateTimeOffset.UtcNow;
```
Comment (petemade) [inline LBH.AdultSocialCare.Api/V1/Domain/DayCarePackageDomains/DayCarePackageDomain.cs:26]: Just for future reference - we should no longer need the date created / date updated fields on our objects as they should now be handled by BaseEntity

File: LBH.AdultSocialCare.Api/V1/Gateways/DayCarePackageOpportunityGateways/DayCarePackageOpportunityGateway.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        {
+            _dbContext = dbContext;
+            _mapper = mapper;
+        }
+        public async Task<Guid> CreateDayCarePackageOpportunity(DayCarePackageOpportunity dayCarePackageOpportunity)
+        {
+            var entry = await _dbContext.DayCarePackageOpportunities.AddAsync(dayCarePackageOpportunity).ConfigureAwait(false);
+            try
+            {
+                await _dbContext.SaveChangesAsync().ConfigureAwait(false);
```
Comment (petemade) [inline LBH.AdultSocialCare.Api/V1/Gateways/DayCarePackageOpportunityGateways/DayCarePackageOpportunityGateway.cs:30]: May be good to check for success of save here, as SaveChangesAsync should return 1 for 1 entity saved

**Review by FurkanKayar55 — APPROVED** [review]

**Review by petemade — APPROVED** [review]

File: LBH.AdultSocialCare.Api/V1/Boundary/DayCarePackageOpportunityBoundary/Request/DayCarePackageOpportunityForUpdateRequest.cs (hunk 1 above)
Comment (dunnudge) [reply LBH.AdultSocialCare.Api/V1/Boundary/DayCarePackageOpportunityBoundary/Request/DayCarePackageOpportunityForUpdateRequest.cs:3]: > I wonder if the creation and update objects can be merged to DayCarePackageOpportunityRequest?

It is possible we can have more fields added to or removed from either the create or update request. Left with a single responsibility that change would be very easy. And also required fields may be different in create and update

**Pull request merged**